pydantic_ai_slim/pydantic_ai/models/openai.py

```diff
@@ -523,6 +526,16 @@ def _process_response(self, response: chat.ChatCompletion | str) -> ModelRespons
         if not isinstance(response, chat.ChatCompletion):
             raise UnexpectedModelBehavior('Invalid response from OpenAI chat completions endpoint, expected JSON data')
 
+        # Some OpenAI-compatible providers (currently only Zhipu/Z.ai, see issue #2723) omit the `object` field even
+        # though the OpenAI schema includes it. We only patch it for that provider to avoid changing validation
+        # error counts in tests that purposefully feed invalid OpenAI responses (which expect 4 errors, including
+        # a missing `object`).
```

Comment:
I don't know exactly what this is referring to, but I'd rather unconditionally set `response.object = 'chat.completion'` here `if response.object is None`, and update the tests to fake invalidity with a different field. So please simplify the check and the commend, similar to the Ollama workaround we have below